apps/studio/src/components/OnboardingRedirect.tsx

```diff
@@ -13,13 +15,27 @@ import {
  */
 const OnboardingRedirect: FunctionComponent = () => {
   const navigate = useNavigate();
+  const dispatch = useAppDispatch();
 
   const { isLoggedIn } = useAppSelector(selectSession);
-  const { data: { firstName, lastName, role } = emptyProfile, isLoading } =
-    useGetProfileQuery(undefined, { skip: !isLoggedIn });
+  const {
+    data: { firstName, lastName, role } = emptyProfile,
+    isLoading,
+    error,
+  } = useGetProfileQuery(undefined, { skip: !isLoggedIn });
 
   useEffect(() => {
-    if (!isLoading && isLoggedIn && (!firstName || !lastName || !role)) {
+    // Exit if still loading or not logged in.
+    if (isLoading || !isLoggedIn) return;
+
+    // If there’s a 401 error (expired token), exit early.
+    if (error && (error as FetchBaseQueryError).status === 401) {
+      dispatch(logOutExpiredSession());
+      return;
+    }
+
+    // If the user’s profile is incomplete, navigate to the create-profile page.
+    if (!firstName || !lastName || !role) {
       navigate("/create-profile");
     }
```

Comment:
As maybe a secondary ticket, should we navigate back to `/home` if the user tries to reach or somehow lands on `/create-profile` when all the details, including location, are already completed?